vault/core.go

```diff
@@ -1115,9 +1152,15 @@ func (c *Core) Seal(token string) (retErr error) {
 	}
 
 	// Validate the token is a root token
-	_, te, err := c.checkToken(logical.UpdateOperation, "sys/seal", token)
+	req := &logical.Request{
+		Operation:   logical.UpdateOperation,
+		Path:        "sys/seal",
+		ClientToken: token,
+	}
+	acl, te, err := c.fetchACLandTokenEntry(req)
```

Comment:
We could, but it required a lot of super hackery because `checkToken` assumes that it is valid to ask whether an ExistenceCheck exists, but the way access control was being handled for `seal`, it was in `RootPaths` in the `system` backend but no actual `logical.Path` existed. It was really an abuse of `RootPaths`.

`checkToken` first calls `fetchACLandTokenEntry`, so we are using the same initial code path; we now just check the ACL directly in `Seal`. This avoids the abuse of `RootPaths` and is really much more understandable and cleaner overall.